go/api/response_handlers.go

```diff
@@ -31,6 +32,72 @@ func handleStringOrNullResponse(response *C.struct_CommandResponse) string {
 	return handleStringResponse(response)
 }
 
+// handleIntegerResponse handles the response from a command that returns an integer.
+func handleIntegerResponse(response *C.struct_CommandResponse) int64 {
+	defer C.free_command_response(response)
+	return int64(response.int_value)
+}
+
+// handleBooleanResponse converts a C struct_CommandResponse's bool_value to a Go bool.
+func handleBooleanResponse(response *C.struct_CommandResponse) bool {
+	defer C.free_command_response(response)
+	return bool(response.bool_value)
+}
+
+// handleStringMapResponse processes a C.struct_CommandResponse to extract an array of Go strings.
+func handleStringMapResponse(response *C.struct_CommandResponse) (map[string]string, error) {
+	defer C.free_command_response(response)
+
+	// Preallocate slice for lengths.
+	lengths := unsafe.Slice(response.array_elements_len, response.array_value_len)
+	numElements := int(response.array_value_len)
+	// Handle empty array case.
+	if numElements == 0 {
+		return nil, nil
+	}
+
+	// Handle inconsistent array lengths (just in case).
+	if numElements%2 != 0 {
+		return nil, errors.New("the number of elements must be even")
+	}
+
+	// Preallocate map for strings with the known size.
+	res := make(map[string]string, numElements/2)
+	values := unsafe.Slice(response.array_value, response.array_value_len)
+
+	for i := 0; i < numElements; i += 2 {
+		res[convertCharArrayToString(values[i], lengths[i])] = convertCharArrayToString(values[i+1], lengths[i+1])
+	}
+
+	return res, nil
+}
+
+func handleStringArrayResponse(response *C.struct_CommandResponse) []string {
+	if response == nil { // Check if the response is nil
+		return nil
+	}
+
+	defer C.free_command_response(response)
+
+	// Preallocate slice for lengths.
+	lengths := unsafe.Slice(response.array_elements_len, response.array_value_len)
+	numElements := int(response.array_value_len)
+	if len(lengths) != numElements {
+		// Handle inconsistent array lengths (just in case)
+		return nil
+	}
+
+	// Preallocate slice for strings with the known size.
+	stringSlice := make([]string, numElements)
+	values := unsafe.Slice(response.array_value, response.array_value_len)
+
+	for i := 0; i < numElements; i++ {
+		stringSlice[i] = convertCharArrayToString(values[i], lengths[i])
+	}
+
+	return stringSlice
+}
+
 func handleLongResponse(response *C.struct_CommandResponse) int64 {
```

Comment:
Add check that `response` is not `nil`
Change this method signature to return the `error` as well